lib/spack/env/cc

```diff
@@ -426,6 +426,12 @@ case "$mode" in
             rpaths+=("${extra_rpaths[@]}")
         fi
 
+        # Set implicit RPATHs
+        IFS=':' read -ra implicit_rpaths <<< "$SPACK_COMPILER_IMPLICIT_RPATHS"
+        if [[ "$add_rpaths" != "false" ]] ; then
+            rpaths+=("${implicit_rpaths[@]}")
+        fi
```

Comment:
Rather than adding a redundant section in `cc`, I think it's easier if the compiler wrappers just have one list of RPATHs, and we deal with the detail that there are two kinds inside `build_environment.py`.  Is there any reason to do the same thing twice here when we could just add these to `SPACK_COMPILER_EXTRA_RPATHS`?

Reply:
This ensures that the user-added paths in `extra_rpaths` are added before the generated ones in `implicit_rpaths`. I'm not sure whether its important though.

Reply:
Yeah my point is that the build_environment can take care of that, since extra_rpaths are ordered.

Reply:
I got halfway through doing this and realized we actually handle them differently. We generate `-L` flags for `extra_rpaths` and not for `implicit_rpaths`.

Reply:
so we should still put them in the order you're saying, I'm just saying it doesn't require additional code in `cc`.

Reply:
It does require additional code in `cc`. The directories from `extra_rpaths` are put in both `libdirs` and `rpaths`, but the directories from `implicit_rpaths` are only put in `rpaths`.

Reply:
Ok this is a good reason to keep them separate. Thanks.